notebooks/input_pipeline_demo.ipynb

```diff
@@ -0,0 +1 @@
ollapsed_sections":[],"authorship_tag":"ABX9TyMELUFuaDJSo2JpVtALibpr"},"kernelspec":{"name":"python3","display_name":"Python 3"}},"cells":[{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"RUQcY5kdrWiO","executionInfo":{"status":"ok","timestamp":1615139842235,"user_tz":480,"elapsed":20765,"user":{"displayName":"Julian Rocha","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14Gi0QrES-WzNCHbIVd-DJKGaCtIC3OO5amuVSCFOPw=s64","userId":"17506259248702060718"}},"outputId":"a825b428-c73c-4777-8133-e1703f85662c"},"source":["# Mount Google Drive\n","from google.colab import drive\n","drive.mount('/content/gdrive')\n","\n","# Change dir to project dir\n","%cd /content/gdrive/MyDrive/Colab\\ Data/COCO-Human-Pose"],"execution_count":null,"outputs":[{"output_type":"stream","text":["Mounted at /content/gdrive\n","/content/gdrive/MyDrive/Colab Data/COCO-Human-Pose\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"Te9ZCKSQ__Zd"},"source":["# (<5 mins)\n","# Setup train/val symlinks for this runtime\n","# puts them in /content/datasets/coco\n","# ensure you have added a Drive shortcut to the dataset in the root of your Drive\n","# this is necessary as the symlinks need a destination that exists\n","# the dataset is hosted here: https://drive.google.com/drive/folders/1EVsLBRwT2njNWOrmBAhDHvvB8qrd9pXT \n","!unzip /content/gdrive/MyDrive/COCO/2017/train2017_symlinks-1000.zip -d /content\n","!unzip /content/gdrive/MyDrive/COCO/2017/val2017_symlinks-1000.zip -d /content\n","\n","!ls /content/datasets/coco/train2017 | wc -l\n","!ls /content/datasets/coco/val2017 | wc -l"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"SBQbXIcUC4sG"},"source":["# (10-20 mins)\n","# Populate the runtimes filesystem cache with dest directories for fast and predictable load times\n","import os\n","def print_files_in_dir(path):\n","  print(len(os.listdir(path)), \" files at \", path)\n","\n","for i in range(0,6):\n","  print_files_in_dir('/content/gdrive/MyDrive/COCO/2017/val2017_sub-1000/' + str(i))\n","\n","for i in range(0,119):\n","    print_files_in_dir('/content/gdrive/MyDrive/COCO/2017/train2017_sub-1000/' + str(i))"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"8eqsEyXetzJV","executionInfo":{"status":"ok","timestamp":1615144372457,"user_tz":480,"elapsed":664,"user":{"displayName":"Julian Rocha","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14Gi0QrES-WzNCHbIVd-DJKGaCtIC3OO5amuVSCFOPw=s64","userId":"17506259248702060718"}}},"source":["from data_generator import DataGenerator\n","\n","# Example Usage of Data Generator\n","train_gen = DataGenerator(\"data/one_ann_min10kp_imgs.csv\",\n","                          \"/content/datasets/coco\",\n","                          (256, 256),\n","                          batch_size=1,\n","                          shuffle=True)\n","# now let's train the model\n","#model.fit(train_gen, epochs=5, ...)"],"execution_count":91,"outputs":[]}]}
```

Comment:
<p>this is great! comments are 10/10</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/robertklee/COCO-Human-Pose/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>